Description: Applied hardening flags to build process.
Author: Alexandre Dantas <alex.dantas92@gmail.com>
Last-Update: 2014-07-31
--- a/Makefile
+++ b/Makefile
@@ -63,8 +63,8 @@
 # Build info
 EXE         = $(PACKAGE)
 CDEBUG      = -O2
-CXXFLAGS    = $(CDEBUG) -Wall -Wextra $(CFLAGS_PLATFORM)
-LDFLAGS     = -lncurses -lyaml-cpp $(LDFLAGS_PLATFORM)
+CXXFLAGS    = $(CDEBUG) -Wall -Wextra $(CFLAGS_PLATFORM) $(shell dpkg-buildflags --get CPPFLAGS)
+LDFLAGS     = -lncurses -lyaml-cpp $(LDFLAGS_PLATFORM) $(shell dpkg-buildflags --get LDFLAGS)
 INCLUDESDIR = -I"src/" -I"deps/"
 LIBSDIR     =
 
@@ -81,7 +81,7 @@
 
 # commander stuff
 COMMANDERDIR = deps/commander
-COMMANDER_CFLAGS = -O2 -Wall -Wextra $(CFLAGS_PLATFORM)
+COMMANDER_CFLAGS = -O2 -Wall -Wextra $(CFLAGS_PLATFORM) $(shell dpkg-buildflags --get CFLAGS)
 COMMANDER_OBJS = $(COMMANDERDIR)/commander.o
 
 # Distribution tarball
